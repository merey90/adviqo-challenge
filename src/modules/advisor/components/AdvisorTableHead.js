import React from 'react';
import {
  TableHead,
  TableCell,
  TableRow,
  TableSortLabel,
  Tooltip
} from '@material-ui/core';

function AdvisorTableHead (props) {
  const { filter, sortHandler } = props;
	console.log("TCL: AdvisorTableHead -> filter", filter)
  
  return (
    <TableHead>
      <TableRow>
        <TableCell>
          Full name
        </TableCell>
        <TableCell>
          Language
        </TableCell>
        <TableCell>
          Status
        </TableCell>
        <TableCell
          sortDirection={filter.sortBy === 'reviews' ? filter.sortDirection : false}
        >
          <Tooltip
            title="Sort"
            enterDelay={300}
          >
            <TableSortLabel
              active={filter.sortBy === 'reviews'}
              direction={filter.sortDirection}
              onClick={() => sortHandler('reviews')}
            >
              Reviews
            </TableSortLabel>
          </Tooltip>
        </TableCell>
      </TableRow>
    </TableHead>
  )
};

export default AdvisorTableHead;