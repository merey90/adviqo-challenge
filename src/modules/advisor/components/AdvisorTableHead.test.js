import React from 'react';
import { shallow } from 'enzyme';

import AdvisorTableHead from './AdvisorTableHead';

const filter = {
  language: 'all',
  status: 'all',
  sortBy: 'reviews',
  sortDirection: 'desc'
};

test('render AdvisorTableHead', () => {
  const wrapper = shallow(
    <AdvisorTableHead
      filter={filter}
      sortHandler={()=>{}}
    />
  );
  expect(wrapper).toMatchSnapshot();
});