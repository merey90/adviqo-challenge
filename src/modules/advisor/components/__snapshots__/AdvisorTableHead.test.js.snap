// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AdvisorTableHead 1`] = `
<AdvisorTableHead
  classes={
    Object {
      "tableCell": "AdvisorTableHead-tableCell-1",
    }
  }
  filter={
    Object {
      "language": "all",
      "sortBy": "reviews",
      "sortDirection": "desc",
      "status": "all",
    }
  }
  sortHandler={[Function]}
/>
`;
